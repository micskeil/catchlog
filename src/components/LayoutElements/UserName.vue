<template>
  <div>
    {{ updatedDisplayName }}
  </div>
</template>

<script>
import { db } from "../../firebase";

export default {
  props: {
    uid: {
      type: String,
      required: true,
    },
  },
  data() {
    return {
      displayName: null,
    };
  },
  computed: {
    updatedDisplayName() {
      return this.displayName;
    },
  },
  methods: {
    getDisplayName() {
      const that = this;
      db.collection("users")
        .doc(this.uid)
        .get()
        .then(function(doc) {
          if (doc.exists) {
            that.displayName = doc.data().displayName;
          } else {
            // doc.data() will be undefined in this case
            console.log("No such document!");
          }
        })
        .catch(function(error) {
          console.log("Error getting document:", error);
        });
    },
  },
  mounted() {
    this.getDisplayName();
  },
};
</script>

<style scoped>
img {
  width: 100%;
  height: 100%;
  border-radius: 100%;
}
</style>
