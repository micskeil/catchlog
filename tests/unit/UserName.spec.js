import { shallowMount } from "@vue/test-utils";
import UserName from "../../src/components/LayoutElements/UserName.vue";

describe("UserName", () => {
  it("should give back the defaultPhoto if no uid prop provided ", () => {
    const propsData = { uid: undefined };
    const wrapper = shallowMount(UserName, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should render to a snapshot when list has items", () => {
    const propsData = { uid: "4FJQKFUuWQdnnWTD7GzGFGvAXZ62" };
    const wrapper = shallowMount(UserName, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });
});
