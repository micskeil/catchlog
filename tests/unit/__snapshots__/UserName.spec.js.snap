// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserName should give back the defaultPhoto if no uid prop provided  1`] = `
<div>
  Guest
</div>
`;

exports[`UserName should render to a snapshot when list has items 1`] = `
<div>
  Guest
</div>
`;
